packages/compass-connections-navigation/src/navigation-item.tsx

```diff
@@ -87,7 +87,7 @@ export function NavigationItem({
     if (item.type === 'connection') {
       return {
         'data-connection-id': item.connectionInfo.id,
-        'data-connection-name': item.connectionInfo.favorite?.name,
+        'data-connection-name': item.name,
```

Comment:
Some connections don't have a favourite name. Not sure if that's intended. But there's always a name because that's what's displayed.